src/sidebar/services/frame-sync.js

```diff
@@ -84,157 +85,156 @@ export class FrameSyncService {
     this._guestRPC = new Bridge();
 
     // Set of tags of annotations that are currently loaded into the frame
-    const inFrame = new Set();
+    this._inFrame = new Set();
 
     /** Whether highlights are visible in guest frames. */
     this._highlightsVisible = false;
 
-    /**
-     * Watch for changes to the set of annotations displayed in the sidebar and
-     * notify connected guests about new/updated/deleted annotations.
-     */
-    this._setupSyncToGuests = () => {
-      let prevPublicAnns = 0;
-
-      watch(
-        store.subscribe,
-        [() => store.allAnnotations(), () => store.frames()],
-        ([annotations, frames], [prevAnnotations]) => {
-          let publicAnns = 0;
-          const inSidebar = new Set();
-          const added = [];
-
-          annotations.forEach(annot => {
-            if (isReply(annot)) {
-              // The frame does not need to know about replies
-              return;
-            }
+    this._setupSyncToGuests();
+    this._setupSyncFromGuests();
+    this._setupHostEvents();
+  }
 
-            if (isPublic(annot)) {
-              ++publicAnns;
-            }
+  /**
+   * Watch for changes to the set of annotations displayed in the sidebar and
```

Comment:
```suggestion
   * Watch for changes to the set of annotations loaded in the sidebar and
```

This code cares about which annotations are loaded, rather than what is _displayed_, which depends on various filtering settings.